### DISCUSSION TITLE
Comparison to Whisper.cpp

### DISCUSSION THREAD

**Discussion by Dev1an** [argmaxinc/WhisperKit#250] [Q&A] [answered]
What sets this project apart from https://github.com/ggerganov/whisper.cpp ?
As far as I can see whisper.cpp does also use Metal and CoreML and in addition also runs on a lot more platforms. What are the advantages of using this project over Whisper.cpp ?

**Comment by atiorh** [comment, marked as answer]
Hi @Dev1an, whisper.cpp is great and each project is taking a very different approach to enabling state-of-the-art on-device ASR. The differences that matter will vary for each use case so we would love to hear what “sets this project apart” for you from you after you evaluate both.

You may find that whisper.cpp is a better fit for you if your only design constraint is to have multi-platform support today. Argmax’s approach is to target and implement one platform at a time:
- Apple support is nearing 1.0 stable release after 9 months of development
- Android support is in 0.1 alpha after 2 months of development
- Other platforms will be announced at a later date

Many developers and enterprises reached out to us for WhisperKit use in production after having tried both projects. The following were among the most frequently quoted factors in their decision:
- Extensible architecture of the pure Swift library design
- End-to-end Core ML support to flexibly deploy to ANE, GPU or CPU for TextDecoder and AudioEncoder components independently
- Advanced features such as Voice Activity Detection, Word Timestamps, and Stable Real-time Streaming and how quickly they were implemented after the initial launch
- High accuracy and low hallucination rates due to quality and latency regression testing across virtually all supported devices
- TestFlight example app as an app integration guide as well as a debugging utility


Some resources to help you map the WhisperKit ecosystem:
- Performance and Quality Benchmarks: https://huggingface.co/spaces/argmaxinc/whisperkit-benchmarks
- Test App: https://testflight.apple.com/join/LPVOyJZW
- WhisperKit Android (Alpha): https://github.com/argmaxinc/WhisperKitAndroid
- whisperkittools: https://github.com/argmaxinc/whisperkittools
